snowflake/src/test/java/org/apache/iceberg/snowflake/SnowflakeCatalogTest.java

```diff
@@ -0,0 +1,231 @@
+            .getBytes());
+    fakeFileIO.addFile(
+        "gs://tab5/metadata/v793.metadata.json",
+        TableMetadataParser.toJson(
+                TableMetadata.newTableMetadata(
+                    schema, partitionSpec, "gs://tab5/", ImmutableMap.<String, String>of()))
+            .getBytes());
+
+    catalog.setFileIO(fakeFileIO);
+
+    Map<String, String> properties = Maps.newHashMap();
+    catalog.initialize(TEST_CATALOG_NAME, properties);
+  }
+
+  @Test
+  public void testListNamespace() {
+    List<Namespace> namespaces = catalog.listNamespaces();
+    Assert.assertEquals(
+        Lists.newArrayList(
+            Namespace.of("DB_1", "SCHEMA_1"),
+            Namespace.of("DB_2", "SCHEMA_2"),
+            Namespace.of("DB_3", "SCHEMA_3"),
+            Namespace.of("DB_3", "SCHEMA_4")),
+        namespaces);
+  }
+
+  @Test
+  public void testListNamespaceWithinDB() {
+    String dbName = "DB_1";
+    List<Namespace> namespaces = catalog.listNamespaces(Namespace.of(dbName));
+    Assert.assertEquals(Lists.newArrayList(Namespace.of(dbName, "SCHEMA_1")), namespaces);
+  }
+
+  @Test
+  public void testListNamespaceWithinNonExistentDB() {
+    // Existence check for nonexistent parent namespaces is optional in the SupportsNamespaces
+    // interface.
+    String dbName = "NONEXISTENT_DB";
+    Assert.assertThrows(RuntimeException.class, () -> catalog.listNamespaces(Namespace.of(dbName)));
+  }
+
+  @Test
+  public void testListNamespaceWithinSchema() {
+    // No "sub-namespaces" beyond database.schema; invalid to try to list namespaces given
+    // a database.schema.
+    String dbName = "DB_3";
+    String schemaName = "SCHEMA_4";
+    Assert.assertThrows(
+        IllegalArgumentException.class,
+        () -> catalog.listNamespaces(Namespace.of(dbName, schemaName)));
+  }
+
+  @Test
+  public void testListTables() {
+    List<TableIdentifier> tables = catalog.listTables(Namespace.empty());
+    Assert.assertEquals(
+        Lists.newArrayList(
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_1"),
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_2"),
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_3"),
+            TableIdentifier.of("DB_2", "SCHEMA_2", "TAB_4"),
+            TableIdentifier.of("DB_3", "SCHEMA_3", "TAB_5"),
+            TableIdentifier.of("DB_3", "SCHEMA_4", "TAB_6")),
+        tables);
+  }
+
+  @Test
+  public void testListTablesWithinDB() {
+    String dbName = "DB_1";
+    List<TableIdentifier> tables = catalog.listTables(Namespace.of(dbName));
+    Assert.assertEquals(
+        Lists.newArrayList(
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_1"),
+            TableIdentifier.of("DB_1", "SCHEMA_1", "TAB_2")),
+        tables);
+  }
+
+  @Test
+  public void testListTablesWithinNonexistentDB() {
+    String dbName = "NONEXISTENT_DB";
+    Assert.assertThrows(RuntimeException.class, () -> catalog.listTables(Namespace.of(dbName)));
```

Comment:
same as above. Also please add a `.hasMessage(..)` check to make sure we're getting the right message here.

Reply:
Done